src/python/turicreate/toolkits/object_detector/_tf_image_augmenter.py

```diff
@@ -407,52 +409,87 @@ def crop_augmenter(image,
 
     return np.array(image), annotation
 
-def complete_augmenter(img_tf, ann_tf, output_height, output_width):
-    img_tf, ann_tf = tf.numpy_function(func=crop_augmenter, inp=[img_tf, ann_tf], Tout=[tf.float32, tf.float32])
-    img_tf, ann_tf = tf.numpy_function(func=padding_augmenter, inp=[img_tf, ann_tf], Tout=[tf.float32, tf.float32])
-    img_tf, ann_tf = tf.numpy_function(func=horizontal_flip_augmenter, inp=[img_tf, ann_tf], Tout=[tf.float32, tf.float32])
-    img_tf, ann_tf = color_augmenter(img_tf, ann_tf)
-    img_tf, ann_tf = hue_augmenter(img_tf, ann_tf)
+def complete_augmenter(img_tf, ann_tf, output_height, output_width, seed_tf):
+    img_tf, ann_tf = crop_augmenter_wrapper(img_tf, ann_tf, seed_tf[0])
+    img_tf, ann_tf = padding_augmenter_wrapper(img_tf, ann_tf, seed_tf[1])
+    img_tf, ann_tf = horizontal_flip_augmenter_wrapper(img_tf, ann_tf, seed_tf[2])
+    img_tf, ann_tf = color_augmenter(img_tf, ann_tf, seed_tf[3])
+    img_tf, ann_tf = hue_augmenter(img_tf, ann_tf, seed_tf[4])
     img_tf, ann_tf = resize_augmenter(img_tf, ann_tf, (output_height, output_width))
+
+    return img_tf, ann_tf
+
+def crop_augmenter_wrapper(img_tf, ann_tf, seed_tf):
+
+    max_attempts = _DEFAULT_AUG_PARAMS["max_attempts"]
+    random_nums = tf.random.uniform([1+4*max_attempts], 0., 1., seed=seed_tf[0])
+    img_tf, ann_tf = tf.numpy_function(func=crop_augmenter, inp=[img_tf, ann_tf, random_nums], Tout=[tf.float32, tf.float32])
     return img_tf, ann_tf
 
+def padding_augmenter_wrapper(img_tf, ann_tf, seed_tf):
+
+    max_attempts = _DEFAULT_AUG_PARAMS["max_attempts"]
+    random_nums = tf.random.uniform([1+max_attempts+3], 0., 1., seed=seed_tf[0])
+    img_tf, ann_tf = tf.numpy_function(func=padding_augmenter, inp=[img_tf, ann_tf, random_nums], Tout=[tf.float32, tf.float32])
+    return img_tf, ann_tf
 
+def horizontal_flip_augmenter_wrapper(img_tf, ann_tf, seed_tf):
+
+    random_nums = tf.random.uniform([1], 0., 1., seed=seed_tf[0])
+    img_tf, ann_tf = tf.numpy_function(func=horizontal_flip_augmenter, inp=[img_tf, ann_tf, random_nums], Tout=[tf.float32, tf.float32])
+    return img_tf, ann_tf
 
 class DataAugmenter(object):
-    def __init__(self, output_height, output_width, batch_size, resize_only):
+    def __init__(self, output_height, output_width, batch_size, resize_only, random_seed=0):
         self.batch_size = batch_size
         self.graph = tf.Graph()
         self.resize_only = resize_only
+
+        #Total num of augmenters with randomness
+        num_augmenter = 5
+
+        #Max num of functions with randomness in one augmenter
+        num_function = 3
+
+        #Generate a random table
+        random_table_shape = (self.batch_size, num_augmenter, num_function)
+        np.random.seed(random_seed if random_seed >=0 else -random_seed)
```

Comment:
Setting the global `numpy` random seed can effect user code that uses `numpy`. Please use `RandomState` instead to avoid clobbering user-level state

https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.random.RandomState.html#numpy.random.RandomState